substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/image/NativeImageDebugInfoProvider.java

```diff
@@ -608,7 +619,11 @@ public int size() {
 
             @Override
             public int modifiers() {
-                return field.getModifiers();
+                ResolvedJavaField targetField = field.wrapped.wrapped;
+                if (targetField instanceof SubstitutionField) {
+                    targetField = ((SubstitutionField) targetField).getOriginal();
```

Comment:
I was not able to test this path with my tests. Recommendations are welcome!

Reply:
Me neither.

Looking at how AnnotationSubstituionProcessor is implemented it looks like you need both a target class and target field labelled with a Substitute annotation (also, probably a TargetElement annotation on the field). If you try using a Substitute annotation on the field when the class just has a TargetClass annotation then you get an error saying you need to use Inject, Delete or Alias on the field. However, I am not 100% clear what the difference is supposed to be between labelling a class with TargetClass vs labelling it with Substitute. I think it is something to do with whether fields that are not explicitly targeted get copied across.

Reply:
> However, I am not 100% clear what the difference is supposed to be between labelling a class with TargetClass vs labelling it with Substitute. I think it is something to do with whether fields that are not explicitly targeted get copied across.

This is covered in https://github.com/oracle/graal/blob/6a8914b68f1709e90c98f9dbd2134d854b2efbf5/substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/annotate/TargetClass.java#L64-L69 seems to cover the last part.

The difference is essentially that in a Class annotated with `TargetClass` you may only `Substitute` the methods you want and the rest of the original methods will be available, on the other hand in a Class annotated with `Substitute` as well all the original methods are deleted and only methods annotated with `Substitute` or `KeepOriginal` are now available.

Reply:
Yeah, that's the comment I was vaguely remembering having read. When I said fields above I really meant to say members.